src/adt/adt.py

```diff
@@ -297,42 +295,52 @@ def _add_memoization(mod, whitelist, ext_key):
         keymap[nm] = id
 
     def create_listkey(f):
-        i = 'i' if f.name != 'i' else 'ii'
-        return (f"tuple( K['{f.type}']({i}) "
-                f"for {i} in {f.name} ),")
+        i = "i" if f.name != "i" else "ii"
+        return f"tuple( K['{f.type}']({i}) " f"for {i} in {f.name} ),"
 
     def create_optkey(f):
-        return (f"None if {f.name} == None else "
-                f"K['{f.type}']({f.name}),")
+        return f"None if {f.name} == None else " f"K['{f.type}']({f.name}),"
```

Comment:
This should probably be `is None`.  Bug from back when I was confused about `is` vs `==`

Reply:
e.g. consider a `bool?`

Reply:
Already changed to `is None` :)